nengo_gui/components/netgraph.py

```diff
@@ -16,74 +16,65 @@
 class NetGraph(Component):
     configs = {}
 
-    def __init__(self, viz, config, uid):
-        super(NetGraph, self).__init__(viz, config, uid)
-        self.viz = viz
-        self.layout = nengo_gui.layout.Layout(self.viz.model)
-        self.config = viz.config
-        self.to_be_expanded = collections.deque([self.viz.model])
+    def __init__(self, sim, config, uid):
+        # this component must be before all the normal graphs (so that
+        # those other graphs are on top of the NetGraph), so its
+        # order is between that of SimControl and the default (0)
+        super(NetGraph, self).__init__(sim, config, uid, component_order=-5)
+        self.sim = sim
+        self.layout = nengo_gui.layout.Layout(self.sim.model)
+        self.to_be_expanded = collections.deque([self.sim.model])
         self.to_be_sent = collections.deque()
         self.uids = {}
         self.parents = {}
-        self.networks_to_search = [self.viz.model]
+        self.networks_to_search = [self.sim.model]
         self.initialized_pan_and_zoom = False
-        try:
-            self.last_modify_time = os.path.getmtime(self.viz.viz.filename)
-        except OSError:
+        if self.sim.filename is None:
             self.last_modify_time = None
+        else:
+            try:
+                self.last_modify_time = os.path.getmtime(self.sim.filename)
+            except OSError:
+                self.last_modify_time = None
         self.last_reload_check = time.time()
 
     def check_for_reload(self):
-        try:
-            t = os.path.getmtime(self.viz.viz.filename)
-        except OSError:
-            t = None
+        if self.sim.filename is not None:
```

Comment:
It's not an OSError if self.sim.filename is None.  Also, I figured it was clearer to explicitly show that I was checking for this, rather than catching TypeError and and adding a comment saying that I'd expect a TypeError in the cases where filename is None (which is what happens all the time in the IPython integration case).